### PR TITLE
error syntax highlighting for vim, highlight

### PR DESCRIPTION
**Pull request by psahabu**
@ben-albrecht is working with error handling and would like syntax highlighting in `vim` sooner. Hence, I'm merging what I have now for error handling syntax in `vim` and `highlight`.

### CODE REVIEW

**Comment by psahabu** [comment]
@lydia-duncan could you take a look at the langDefs changes here?

File: highlight/highlight/langDefs/chpl.lang [hunk 1]
```diff
@@ -11,16 +11,16 @@ Description="Chapel"
...
 Keywords={
   { Id=1,
-   List={"as", "align", "atomic", "begin", "break", "by", "class", "cobegin",
-         "coforall", "config", "const", "continue", "proc", "iter", "delete",
-         "dmapped", "do", "domain", "else", "enum", "except", "export",
-         "extern", "false", "for", "forall", "if", "in", "index", "inline",
-         "inout", "label", "lambda", "let", "local", "module", "new", "nil",
-         "noinit", "on", "only", "otherwise", "out", "param", "private",
+   List={"as", "align", "atomic", "begin", "break", "by", "catch", "class",
+         "cobegin", "coforall", "config", "const", "continue", "proc", "iter",
```
Comment (lydia-duncan) [inline highlight/highlight/langDefs/chpl.lang]: It bothers me that proc and iter are not alphabetically sorted, but that isn't your fault (and I am sad I didn't notice it before now)

**Comment by lydia-duncan** [comment]
No other concerns, you can take or leave my one comment :)

**Comment by psahabu** [comment]
@lydia-duncan: Done.

@ben-albrecht: It seems that [Swift's syntax highlighting doesn't bother](https://github.com/apple/swift/blob/master/utils/vim/syntax/swift.vim) with highlighting the `!` in `try!` in vim. Should we just merge this as is then?

**Comment by ben-albrecht** [comment]
> It seems they use ! as an operator in some circumstances.

Makes sense. I'd be in favor of dropping `try!` from the vim highlighting and accept that `!` will not be highlighted.

**Comment by psahabu** [comment]
@ben-albrecht Sounds good, and after consulting with Lydia I have also dropped `try!` from langDefs least until we need to update it again.

**Pull request merged**